core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EvaluationStatistics.java

```diff
@@ -82,6 +85,8 @@ public void meet(Slice slice) {
 
 		protected double getCardinality(StatementPattern pattern) {
 			List<Var> vars = pattern.getVarList();
+			Set<Var> varSet = new HashSet<Var>(vars);
+			vars = new ArrayList<Var>(varSet);
```

Comment:
I think you can also just do:

    Set<Var> vars = new HashSet<>(pattern.getVarList());

I don't think the rest of the method requires that `vars` is an ArrayList (or even a list), so no need to create an arraylist back out of the set. Saves having to copy the elements over twice.

Reply:
Alright! I will make this change and update this pull request.